### PR TITLE
[WIP][data] feat: TransferQueue - integrate TransferQueue into main codebase

### PR DESCRIPTION
**Pull request by 0oshowero0**
### What does this PR do?

This PR integrates the experimental `TransferQueue` feature into the main codebase. 

### 1. New Trainer Implementation
- Added `RayPPOTrainerTransferQueue` which inherits from `RayPPOTrainer`.
- Implemented `BatchMeta`-based operations to replace the distribution of real data objects (`DataProto/TensorDict`).

### 2. Configuration
- Updated `ppo_trainer.yaml` and `ppo_megatron_trainer.yaml` to include TransferQueue-related configurations.

### 3. Integration & Entry Point
- Modified `RayPPOTrainerTransferQueue` to adapt to the new features.
- Added switch logic in `main_ppo.py` to enable the TransferQueue via config.\
- Update CI & scripts of TQ

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: verl/workers/engine_workers.py [hunk 1]
```diff
@@ -538,31 +553,45 @@ def init_model(self):
...
-    @register(dispatch_mode=make_nd_compute_dataproto_dispatch_fn(mesh_name="ref"))
+    @register(
+        dispatch_mode=make_nd_compute_dataproto_dispatch_fn(mesh_name="ref"), put_data=True, convert_type="TensorDict"
+    )
     @DistProfiler.annotate(color="olive", role="ref_compute_log_prob")
     def compute_ref_log_prob(self, data: TensorDict) -> TensorDict:
-        output = self.ref.infer_batch(data=data)
+        # TODO(TQ): need to remove the pad and unpad processing
+        if self.tq_config is not None and self.tq_config["enable"]:
+            data["loss_mask"] = data["response_mask"]
```
Comment (0oshowero0) [inline verl/workers/engine_workers.py]: be careful with these operations

File: verl/workers/utils/losses.py [hunk 2]
```diff
@@ -72,28 +75,44 @@ def _slice_response_from_unpad_output(tensor: torch.Tensor, data: TensorDict) ->
...
+            #     f"Length mismatch: tensor={nt.size(0)}, resp_len={resp_len}"
+            pad_size = max_response_len - resp_len
+            response_list.append(F.pad(nt, (0, pad_size)))
+        output = torch.stack(response_list, dim=0)
     return output
 
 
 def ppo_loss(config: ActorConfig, model_output, data: TensorDict, dp_group=None):
+    """
+    Convert tensors into padded ones even for TQ version.
```
Comment (0oshowero0) [inline verl/workers/utils/losses.py]: can we move this padding process even earlier, thus preventing changing loss related stuff?

File: verl/experimental/agent_loop/agent_loop.py [hunk 3]
```diff
@@ -180,6 +186,84 @@ class _InternalAgentLoopOutput(AgentLoopOutput):
...
+            non_tensor_data["__num_turns__"] = [self.num_turns]
+        # 3.4 metrics
+        if self.metrics is not None:
+            metrics = self.metrics.model_dump()
+            meta_info["metrics"] = [metrics]
+
+        # 4. extra_fields
+        extra_fields = {}
+        all_keys = set(key for key in self.extra_fields)
+        for key in all_keys:
```
Comment (Chase-Rong) [inline verl/experimental/agent_loop/agent_loop.py]: for key in self.extra_fields ?

File: verl/experimental/agent_loop/agent_loop.py [hunk 4]
```diff
@@ -413,6 +506,20 @@ async def generate_sequences(self, batch: DataProto) -> DataProto:
             responses:     |<- LLM generation ->|<- tool_calls ->|<- LLM generation ->|<- padding ->|
             response_mask: | 1, 1, 1, ..., 1, 1 | 0, 0, .., 0, 0 | 1, 1, 1, ..., 1, 1 | 0, 0, ..., 0|
         """
+        if isinstance(batch, BatchMeta) and self.tq_client is not None:
+            batch_meta = batch
+            batch = await self.tq_client.async_get_data(batch_meta)
+            meta_info = batch_meta.get_all_extra_info()
+            non_tensor_batch = {}
+            for key, val in batch.items():
```
Comment (0oshowero0) [inline verl/experimental/agent_loop/agent_loop.py:513]: can be a general util

File: verl/workers/utils/losses.py [hunk 5]
```diff
@@ -54,8 +54,11 @@ def sft_loss(config: ActorConfig, model_output, data: TensorDict, dp_group=None)
     return loss, {}
 
 
-def _slice_response_from_unpad_output(tensor: torch.Tensor, data: TensorDict) -> torch.Tensor:
-    """Slice response from unpad model output.
+def _slice_response_from_unpad_output(
```
Comment (0oshowero0) [inline verl/workers/utils/losses.py]: TODO：do not modify this
